### PR TITLE
Sample Jaeger version 1.20 -> 1.22

### PR DESCRIPTION
**Pull request by esnible**
Upgrade the Jaeger sample to 1.22.

Jaeger 1.22 was released in February.
https://github.com/jaegertracing/jaeger/releases/tag/v1.22.0

Key changes
- When Jaeger went from 1.21 to 1.22 they changed a CLI flag we set via Viper.  This means it isn't sufficient for our users to just change the Docker image.  If you are like me, and don't read the release notes, it is confusing
- Istio sends traces to Jaeger using Zipkin protocol.  I found it confusing that we expose Zipkin port 9411 on the zipkin.istio-system service but not on the jaeger-collector service.  Users should be able to set `meshConfig.defaultConfig.tracing.zipkin.address` to either `jaeger-collector.istio-system.svc.cluster.local:9411` or `zipkin.istio-system:9411`.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by bianpengyuan — APPROVED** [review]

**Comment by douglas-reid** [comment]
Not exposing the jaeger-collector service was intentional. Since we are sending via Zipkin API, we want the traffic to go to the Zipkin service, which can be backed by the jaeger collector workloads.  The `zipkin` service is meant to be de-coupled from the backing implementation. I'm not sure that there is any real harm in adding the port to collector service in the sample, but it will lead to questions about why the jaeger sample even includes the `zipkin` service. The Zipkin community has fairly strong feelings about proper attribution of APIs. It be nice to avoid venturing back into those waters, and maintaining docs that suggest zipkin API requests go to the zipkin service.

File: samples/addons/jaeger.yaml [hunk 1]
```diff
@@ -68,6 +68,10 @@ spec:
       port: 80
       protocol: TCP
       targetPort: 16686
+    - name: http2-query
```
Comment (jpkrohling) [inline samples/addons/jaeger.yaml:71]: Would `grpc-query` perhaps be more appropriate?

Comment (esnible) [reply samples/addons/jaeger.yaml:71]: @jpkrohling I will change this to say 

```yaml
    # Note: Change port name if you add '--query.grpc.tls.enabled=true'
    - name: grpc-query
```

For the kinds of things I am doing -- `kubectl port-forward`-ing and using _grpcurl_ from my laptop the port name won't matter.  Meshed pods should be able to query Jaeger (e.g. if I want to build a tool that uses Jaeger's jaeger.api_v2.QueryService gRPC service).  See https://github.com/istio/istio/issues/16282#issuecomment-665349098 for why the name can't start with `grpc` if gRPC is using TLS.

**Comment by esnible** [comment]
> Not exposing the jaeger-collector service was intentional. Since we are sending via Zipkin API, we want the traffic to go to the Zipkin service, which can be backed by the jaeger collector workloads. The `zipkin` service is meant to be de-coupled from the backing implementation.

@douglas-reid This makes sense.  I added the Zipkin port to the Jaeger Collector Service because https://istio.io/latest/docs/ops/integrations/jaeger/#option-2-customizable-install says

```
This can be configured with --set values.global.tracer.zipkin.address=<jaeger-collector-address>:9411 at installation time.
```

Because I read this, I got the notion into my head that I could use `jaeger-collector:9411`.  If we want the user to use `zipkin:9411`, the docs should make it clear that a users Jaeger collector might be exposed under different service names.

The Jaeger Operator creates a 9411 on the collector called `http-zipkin`.  This provides some attribution.  If I make Istio's same Jaeger Collector name the port `http-zipkin`, is that enough attribution for us?

File: samples/addons/jaeger.yaml (hunk 1 above)
Comment (howardjohn) [reply samples/addons/jaeger.yaml]: Not sure I understand. `http2` and `grpc` should be the same, if not its a bug. Let me know if its not working with either

Comment (howardjohn) [reply samples/addons/jaeger.yaml]: The linked issue is about using TLS with a port named "grpc". TLS with a port named "http2" is equally invalid

**Pull request merged**